protocols/request-response/src/lib.rs

```diff
@@ -470,20 +470,14 @@ where
     ///
     /// Returns true if the address was added, false otherwise (i.e. if the
     /// address is already in the list).
+    #[deprecated(note = "Use `Swarm::add_peer_address` instead.")]
     pub fn add_address(&mut self, peer: &PeerId, address: Multiaddr) -> bool {
-        self.addresses.entry(*peer).or_default().insert(address)
+        self.addresses.add(*peer, address)
     }
 
-    /// Removes an address of a peer previously added via `add_address`.
+    /// Removes an address of a peer previously added via [`Behaviour::add_address`].
     pub fn remove_address(&mut self, peer: &PeerId, address: &Multiaddr) {
```

Comment:
We need to deprecate this one too. You can say in the message that it will be removed without replacement.

Reply:
:heavy_check_mark: